src/NuGet.Clients/NuGet.PackageManagement.UI/PackageItemLoader.cs

```diff
@@ -75,9 +73,14 @@ public LoadingStatus LoadingStatus
 
             // returns the "raw" counter which is not the same as _results.Items.Count
             // simply because it correlates to un-merged items
-            public int ItemsCount => _results?.RawItemsCount ?? 0;
+            public int ItemsCount => Results?.RawItemsCount ?? 0;
+
+            public IDictionary<string, LoadingStatus> SourceLoadingStatus => Results?.SourceSearchStatus;
 
-            public IDictionary<string, LoadingStatus> SourceLoadingStatus => _results?.SourceSearchStatus;
+            public void Dispose()
+            {
+                Results = null;
+            }
```

Comment:
If this class is never inherited, mark it as sealed, otherwise [implement the dispose pattern](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implement-the-dispose-pattern)

Also, what's the purpose of making this disposable?  `Dispose` isn't calling `Dispose` on any properties or fields, and the garbage collector will dispose `Results`'s instance.

Reply:
Removed.